lib/accumulate_distribute/meta/init_state.js

```diff
@@ -17,19 +18,20 @@ const genOrderAmounts = require('../util/gen_order_amounts')
  * @param {object} args - instance execution parameters
  * @returns {object} initialState
  */
-const initState = (args = {}) => {
+const initState = (args = {}, pairConfig = {}) => {
   const { amount } = args
-  const orderAmounts = genOrderAmounts.gen(args)
+  const orderAmounts = genOrderAmounts.gen(args, pairConfig)
 
   let totalAmount = 0
-  orderAmounts.forEach(a => { totalAmount += a })
+  orderAmounts.forEach(a => { totalAmount = +prepareAmount(totalAmount + a) })
 
-  if (Math.abs(totalAmount - amount) > DUST) {
+  if (Math.abs(totalAmount) - Math.abs(amount) > DUST) {
```

Comment:
please use nBN instead of classic math calcs to avoid floating prec issues

Reply:
hmm .. i wanted to .. but tried to minimize usage of big number as much as possible since it takes up a significant time for even simple operation.. the float precision issue seem to be handled by DUST here .. so didn't think it was necessary to use big number here